src/WingetCreateCLI/PromptHelper.cs

```diff
@@ -0,0 +1,270 @@
+        /// </summary>
+        /// <param name="model">Object model to be modified.</param>
+        /// <param name="memberName">Name of the selected property field.</param>
+        /// <param name="minimum">Specifies the minimum number of entries if the property is a list.</param>
+        /// <param name="validationModel">Object model to be validated against if the target field differs from what is specified in the model (i.e. NewCommand.InstallerUrls).</param>
+        /// <param name="validationName">Name of the property field to be used for looking up validation constraints if the target field name differs from what specified in the model.</param>
+        public static void PromptAndSetPropertyValue(object model, string memberName, int minimum = 0, object validationModel = null, string validationName = null)
+        {
+            if (string.IsNullOrEmpty(validationName))
+            {
+                validationName = memberName;
+            }
+
+            if (validationModel == null)
+            {
+                validationModel = model;
+            }
+
+            string message = string.Format(Resources.FieldValueIs_Message, memberName);
+            Console.WriteLine(Resources.ResourceManager.GetString($"{memberName}_KeywordDescription"));
+            var property = model.GetType().GetProperty(memberName);
+            var defaultValue = property.GetValue(model);
+            Type propertyType = property.PropertyType;
+            Type elementType;
+
+            if (propertyType == typeof(string))
+            {
+                var result = Prompt.Input<string>(message, property.GetValue(model), new[] { FieldValidation.ValidateProperty(validationModel, validationName, defaultValue) });
+                property.SetValue(model, result);
+            }
+            else if (propertyType.IsEnum)
+            {
+                PromptEnum(model, property, defaultValue.GetType(), true, message, defaultValue.ToString());
+            }
+            else if ((elementType = Nullable.GetUnderlyingType(propertyType)) != null)
+            {
+                if (elementType.IsEnum)
+                {
+                    PromptEnum(model, property, elementType, false, message: message);
+                }
+            }
+            else if (propertyType.IsList())
+            {
+                elementType = propertyType.GetGenericArguments().SingleOrDefault();
+                if (elementType == typeof(string) || typeof(IEnumerable<string>).IsAssignableFrom(propertyType))
+                {
+                    var value = Prompt.List<string>(message, minimum: minimum, validators: new[] { FieldValidation.ValidateProperty(validationModel, validationName, defaultValue) });
+                    if (!value.Any())
+                    {
+                        value = null;
+                    }
+
+                    property.SetValue(model, value);
+                }
+                else if (elementType == typeof(int))
```

Comment:
Can you make this more generic? What if the property was a double, or long, etc?

Reply:
Made generic